src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -633,7 +632,50 @@ class World internal constructor(
      */
     fun dispose() {
         entityService.removeAll()
-        systems.forEachReverse { it.onDispose() }
+        systems.reversed().forEach { it.onDispose() }
+    }
+
+    internal fun setUpAggregatedFamilyHooks(systems: List<IntervalSystem>) {
```

Comment:
I think this functionality is not properly working anymore with the new change. It was a very special feature requested by some other user. To explain it:
- In Fleks you can have family hooks to react when an entity gets added/removed from a family
- To make life easier we added a `FamilyOnAdd` and `FamilyOnRemove` interface that can be added to an `Iteratingsystem` and then Fleks will automatically register those hooks to the family
- Since there is just one hook on a family we then simply combined all system hook functions into a single one and register that special function as a hook to the family
- One additional point that makes it a little complex is that you can also register your own family hook in the world configuration already (that's the `ownHook` in the code below. The final hook is then the family original hook + all systems' hooks

Before, this was just necessary once because systems could not be modified after world creation. Now, imo we need to do that all the time when a system gets added or removed.

However, it is not necessary to iterate over all systems all the time and create the hooks again. My idea is:
- When adding/removing a system check if it implements the `FamilyOnAdd` or `FamilyOnRemove` interface 
- If it does then call this method or a new one which does the following:
  - get all systems that have the same family in their current order
  - update the family hook with the already existing logic
  - This will add/remove the newly added/removed system's hooks

However, there is one issue with the `ownHook` of the family. I think we don't have that anymore and it is currently impossible to retrieve it from somewhere. Needs to be checked but I am pretty sure that that's the case, unfortunately :(

For that I see two solutions:
- store the own hook of the family, that gets created in the world configuration in a separate place somewhere where we can then use it for the logic above (e.g. directly in the family class)
  - The downside is that we introduce an additional field per family which increases memory consumption but I guess it is not huge
  - This could maybe be solved more efficiently if we e.g. use a `ownHookCache` in the world directly. When we retrieve a family`s ownHook for the first time we could read this cache. If there is an entry then use it and if there is no entry then add it. That way we don't add an additional field to every family.
- instead of just having one hook on a family just make it a list or an array. That way we don't need to override the hook at all and just add/remove system hooks.
  - This is however inconsistent with the other hooks in Fleks